core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -76,6 +76,8 @@ class SparkEnv (
   // (e.g., HadoopFileRDD uses this to cache JobConfs and InputFormats).
   private[spark] val hadoopJobMetadata = new MapMaker().softValues().makeMap[String, Any]()
 
+  private var tmpFilesDir: Option[String] = None
```

Comment:
Yeah this looks good. Maybe call it `driverTmpDirToDelete` or something to be extra clear?